ext/droption/droption.h

```diff
@@ -711,24 +717,15 @@ inline bool
 droption_t<unsigned long>::convert_from_string(const std::string s)
 {
     errno = 0;
-    long input = strtol(s.c_str(), NULL, 10);
-    if (input >= 0)
-        value_ = (unsigned long)input;
-    else
-        return false;
```

Comment:
Don't we need to return false on negative inputs?

Reply:
No signed type contains all unsigned long and unsigned long long values. So if we want to check whether it is a negative value, the best way is to check whether the first valid character is '-'. Do we need to check it?

Reply:
> I think we can set the option's max and min to avoid the out-of-range issue.

So the suggestion is: if negative values are invalid for some option, the user should set the appropriate minval_ so that it fails in clamp_value, instead of failing in parse_from_string.
- I see that a failure in clamp_value also fails parse_argv (same as a failure in parse_from_string). So droptions remains the same functionally.
- Also, we're making the non-negative requirement more explicit, which is good.

Sounds good to me. @derekbruening?

Reply:
If this is for `droption_t<unsigned long>` the minval cannot be negative anyway as it's unsigned.  So the real issue is that this unsigned value type parsing should use strtoul instead of strtol, right?

Reply:
Not really. The issue is whether we accept parsing a negative value to an unsigned value. 
Because strtoul() can parse the negative value to the unsigned value, it can parse -1 to UINT64_MAX. 
We allow the caller to define their valid range in droption. So we don't need negative value checking. Because 
For example, if they set to only accept values from 1 to 2^32. When someone types an -1, droption can directly give an issue that it is out of range. If someone types -2^32-1, strtoul() can output 2^32-1, and the value is valid.